app/mailboxes/application_mailbox.rb

```diff
@@ -37,10 +37,24 @@ def in_reply_to_matches?(in_reply_to)
     # checks if follow this pattern  send it to reply_mailbox
     # reply+<conversation-uuid>@<mailer-domain.com>
     def reply_uuid_mail?(inbound_mail)
+      validate_to_address(inbound_mail)
+
       inbound_mail.mail.to&.any? do |email|
         conversation_uuid = email.split('@')[0]
         conversation_uuid.match?(REPLY_EMAIL_UUID_PATTERN)
       end
     end
+
+    # if mail.to returns a string, then it is a malformed `to` header
+    # valid `to` header will be of type Mail::AddressContainer
+    # validate if the to address is of type string
+    def validate_to_address(inbound_mail)
+      to_address_class = inbound_mail.mail.to&.class
+
+      return unless to_address_class == String
```

Comment:
Shouldn't we validate the type Mail::AddressContainer`?

Reply:
@vishnu-narayanan Can you please share an example header that is malformed?

Reply:
> Shouldn't we validate the type Mail::AddressContainer`?

@scmmishra yeah, that should also work and be more readable. Let me make the change.

Reply:
> Can you please share an example header that is malformed?

@pranavrajs Sure, please see the comment below. This example fixture file is from an actual failed email in chatwoot prod though.